### PR TITLE
Apply downstream patches from Debian

### PR DESCRIPTION
**Pull request by bradking**
The Debian package for 1.23.0 builds with [these patches](https://salsa.debian.org/debian/libuv1/tree/debian/1.23.0-2/debian/patches).  Apply those that are still relevant to upstream v1.x.

I've authored commit messages with relevant information about the purpose of each patch.

Please note that the two PATH_MAX patches go together but I left them separate for authorship and explanation purposes.

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -389,12 +390,31 @@ static ssize_t uv__fs_pathmax_size(const char* path) {
...
   ssize_t len;
   char* buf;
+  struct stat st;
+  int ret;
+
+  ret = lstat(req->path, &st);
+  if (ret != 0) {
+    return -1;
+  }
+  len = st.st_size;
```
Comment (addaleax) [inline src/unix/fs.c]: This seems like it could potentially lead to a very large allocation if the target file is not actually a symlink?

**Comment by bradking** [comment]
!2008 supersedes the PATH_MAX patches.

**Comment by bnoordhuis** [comment]
The second and third commit LGTM. If your intent is to drop the first and the last commit, can you do that so I can sign off on the PR as a whole? Thanks.

**Comment by bradking** [comment]
I've dropped the path_max commits in favor of #2008.  The rest of the commits here are independent.

**Comment by santigimeno** [comment]
Landed in https://github.com/libuv/libuv/commit/c5d2bf12f7565e143b6becd2560428e602485d06 and https://github.com/libuv/libuv/commit/1d1969adcc90b6b5f7e9f39a7396c9a619057158. Thanks!

**Pull request closed without merging**